doc/examples/fast_streamline_search.py

```diff
@@ -0,0 +1,203 @@
+
+"""
+Download and read data for this tutorial
+"""
+
+hcp_file = get_target_tractogram_hcp()
+streamlines = load_trk(hcp_file, "same", bbox_valid_check=False).streamlines
+
+"""
+Visualize the atlas (ref) bundle and full brain tractogram
+"""
+
+interactive = False
+
+scene = window.Scene()
+scene.SetBackground(1, 1, 1)
+scene.add(actor.line(streamlines))
+if interactive:
+    window.show(scene)
+else:
+    window.record(scene, out_path='tractograms_initial.png', size=(600, 600))
+
+"""
+.. figure:: tractograms_initial.png
+   :align: center
+
+   Atlas bundle and source streamlines before registration.
+
+"""
+
+"""
+Read Arcuate Fasciculus Left and Corticospinal Tract Left bundles from already
+fetched atlas data to use them as model bundle. Let's visualize the
+Arcuate Fasciculus Left model bundle.
+"""
+
+model_af_l_file, model_cst_l_file = get_two_hcp842_bundles()
+sft_af_l = load_trk(model_af_l_file, "same", bbox_valid_check=False)
+model_af_l = sft_af_l.streamlines
+
+scene = window.Scene()
+scene.SetBackground(1, 1, 1)
+scene.add(actor.line(model_af_l, colors=(0, 1, 0)))
+scene.set_camera(focal_point=(-18.17281532, -19.55606842, 6.92485857),
+                 position=(-360.11, -30.46, -40.44),
+                 view_up=(-0.03, 0.028, 0.89))
+if interactive:
+    window.show(scene)
+else:
+    window.record(scene, out_path='AF_L_model_bundle.png',
+                  size=(600, 600))
+
+"""
+.. figure:: AF_L_model_bundle.png
+   :align: center
+
+   Model Arcuate Fasciculus Left bundle
+
+"""
+
+"""
+Search for all similar streamlines  [StOnge2021]_
+
+Fast Streamline Search can do a radius search
+to find all streamlines that are similar to from one tractogram to another.
+It return the distance matrix mapping between both tractograms.
+The same list of streamlines can be given to recover the self distance matrix.
+
+    FastStreamlinesSearch Class
+    - ref_streamlines : reference streamlines, that will be searched in (tree)
+    - max_radius : is the maximum distance that can be used with radius search
+
+    radius_search() function
+    - radius : for each streamline search find all similar ones in the
+        "ref_streamlines" that are within the given radius
+
+    Be cautious, a large radius might result in a dense distance computation,
+    requiring a large amount of time and memory.
+    Recommended range of the radius is from 1 to 10 mm.
+"""
+
+radius = 7.0
+fs_tree_af = FastStreamlineSearch(ref_streamlines=model_af_l, max_radius=radius)
+coo_mdist_mtx = fs_tree_af.radius_search(streamlines, radius=radius)
+
+
+"""
+Extract indices of streamlines with an similar ones in the reference
+"""
+ids_s = np.unique(coo_mdist_mtx.row)
+ids_ref = np.unique(coo_mdist_mtx.col)
+
+recognized_af_l = streamlines[ids_s]
+
+"""
+let's visualize streamlines similar to the Arcuate Fasciculus Left bundle
+"""
+
+scene = window.Scene()
+scene.SetBackground(1, 1, 1)
+scene.add(actor.line(model_af_l, colors=(0, 1, 0)))
```

Comment:
I suggest removing the model from this figure

Reply:
done